llvm/lib/YkIR/YkIRWriter.cpp

```diff
@@ -479,6 +480,29 @@ class YkIRWriter {
     }
   }
 
+  void serialiseIndirectCallInst(CallInst *I, FuncLowerCtxt &FLCtxt,
+                                 unsigned BBIdx, unsigned &InstIdx) {
+
+    serialiseOpcode(OpCodeIndirectCall);
+    // function type:
+    OutStreamer.emitSizeT(typeIndex(I->getFunctionType()));
+    // callee (operand):
+    serialiseOperand(I, FLCtxt, I->getCalledOperand());
+    // num_args:
+    // (this includes static and varargs arguments)
+    OutStreamer.emitInt32(I->arg_size());
```

Comment:
This doesn't need to happen in this PR, but we should find a way to check if the various `num_` fields we serialise actually fit the data that we populate them with.

e.g. here, does `arg_size()` always fit in 32-bits? etc.

Reply:
(this has been on my mind a few times recently)